src/plugins/data/public/ui/dataset_selector/dataset_table.tsx

```diff
@@ -86,7 +86,27 @@ export const DatasetTable: React.FC<DatasetTableProps> = (props) => {
       <EuiBasicTable
         items={dataStructures}
         itemId="id"
-        columns={[{ field: 'title', name: 'Name' }]}
+        columns={[
+          {
+            field: 'title',
+            name: 'Name',
+            textOnly: true,
+            render: (title: string, item: DataStructure) => {
+              return (
+                <>
+                  <EuiText size="s" className="datasetTable__itemTitle">
+                    {title}
+                  </EuiText>
+                  {item.description && (
+                    <EuiText size="s" color="subdued" className="datasetTable__itemDescription">
+                      {item.description}
```

Comment:
May be add ellipsis and truncation to support longer descriptions? or it is that dataset type config would have this logic?

Reply:
added https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9296#discussion_r1945287420